### PR TITLE
Gmarchetti/flexible env vars for controller

### PR DESCRIPTION
**Pull request by galenmarchetti**
This allows writers of test controllers to pass in flexible environment variables from the initializer to the controller. This is particularly important for passing in arbitrary image names for example for byzantine tests

### CODE REVIEW

File: initializer/parallelism/test_executor.go [hunk 1]
```diff
@@ -271,4 +276,8 @@ func generateTestControllerEnvVariables(
 		testVolumeArg:           testVolumeName,
 		testVolumeMountpointArg: testVolumeMountpoint,
 	}
+	for key, val := range(envVars) {
+		standardVars[key] = val
```
Comment (mieubrisse) [inline initializer/parallelism/test_executor.go:289]: We should def have error-checking here to make sure that the user doesn't unknowingly pass in an argument that quashes one of our special environment variables!

File: initializer/parallelism/test_executor_parallelizer.go [hunk 2]
```diff
@@ -52,14 +53,16 @@ func NewTestExecutorParallelizer(
 			testControllerImageName string,
 			testControllerLogLevel string,
 			testServiceImageName string,
+			testControllerEnvVars map[string]string,
```
Comment (mieubrisse) [inline initializer/parallelism/test_executor_parallelizer.go:57]: Let's get in the habit of updating function docs whenever a function signature is modified!

File: initializer/parallelism/test_executor.go [hunk 3]
```diff
@@ -258,8 +262,9 @@ func generateTestControllerEnvVariables(
 			testName string,
 			logLevel string,
 			testServiceImageName string,
-			testVolumeName string) map[string]string {
-	return map[string]string{
+			testVolumeName string,
+			envVars map[string]string) map[string]string {
```
Comment (mieubrisse) [inline initializer/parallelism/test_executor.go]: Same deal - needs function docs update!

**Review by mieubrisse — APPROVED** [review]
Approved pending comments being fixed

File: initializer/parallelism/test_executor.go (hunk 1 above)
Comment (galenmarchetti) [reply initializer/parallelism/test_executor.go:289]: good shout! implementing

**Pull request merged**